### PR TITLE
updated automod

### PR DESCRIPTION
**Pull request by Commandserver**
https://discord.com/developers/docs/resources/auto-moderation

### CODE REVIEW

File: include/dpp/automod.h [hunk 1]
```diff
@@ -83,10 +83,6 @@ enum automod_trigger_type : uint8_t {
 	 * @brief Keyword filtering
 	 */
 	amod_type_keyword = 1,
-	/**
-	 * @brief Harmful/malware links
-	 */
-	amod_type_harmful_link = 2,
```
Comment (Commandserver) [inline include/dpp/automod.h:89]: removed by discord

File: include/dpp/automod.h [hunk 2]
```diff
@@ -95,54 +91,45 @@ enum automod_trigger_type : uint8_t {
...
-     * | \*the mat* | brea<u><b>the mat</b></u>ter          |
-     *
-     * Whole Word - keyword is a full word or phrase and must be surrounded by whitespace at the beginning and end
-     *
-     * | keyword | matches     |
-     * |---------|-------------|
-     * | cat     | <u><b>Cat</b></u>     |
-     * | the mat | <u><b>the mat</b></u> |
-     *
+	 * @see Keyword matching strategies: https://discord.com/developers/docs/resources/auto-moderation#auto-moderation-rule-object-keyword-matching-strategies
```
Comment (Commandserver) [inline include/dpp/automod.h:111]: i replaced the fat comment with a mention to their docs

Comment (braindigitalis) [reply include/dpp/automod.h:111]: the comment was there so people didnt have to click through to discords docs.
Discords docs are very generic, and i think this has a place reused on dpp.dev.

Comment (Commandserver) [reply include/dpp/automod.h]: fair enough. ill put it back in ;)

File: include/dpp/application.h [hunk 3]
```diff
@@ -60,7 +60,9 @@ enum application_flags : uint32_t {
 	/// Has approval for message content
 	apf_gateway_message_content = (1 << 18),
 	/// Has message content, but <100 guilds
-	apf_gateway_message_content_limited = (1 << 19)
+	apf_gateway_message_content_limited = (1 << 19),
+	/// Indicates if the app has registered global application commands
+	apf_application_command_badge = (1 << 23)
```
Comment (Commandserver) [inline include/dpp/application.h:65]: https://github.com/discord/discord-api-docs/commit/96bd1882cdd1113ce9688a1922ef4952bf1fec90

Comment (braindigitalis) [reply include/dpp/application.h:65]: looks good to me 👍

File: include/dpp/automod.h (hunk 1 above)
Comment (braindigitalis) [reply include/dpp/automod.h:89]: fair play, is it completely removed and not marked deprecated?

Comment (Commandserver) [reply include/dpp/automod.h:89]: discord removed it completely from their docs. i've re added it here and marked it as deprecated.

**Pull request merged**